docs/DeveloperGuide.md

```diff
@@ -1,37 +1,176 @@
 # Developer Guide
 
-## Acknowledgements
-
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
-
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Results feature
+
+The results feature is facilitated by `ResultsList` and 
+`AnswerTracker`. Both are used to display the results for
+all question sets attempted by the user, including details
+such as specific questions and their respective answer inputs
+from the user.
+
+Given below is an example usage scenario and how the results
+mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time,
+and proceeds to start a game with their chosen topic.
+
+The following sequence diagram shows how the `Results` for
+one question set is added to the `ResultsList`:  
+![ResultsList sequence diagram](./team/img/Results.png)
+> **Note:** The lifeline for Parser and Results should end
+> at the destroy marker (X) but due to a limitation of PlantUML,
+> the lifeline reaches the end of the diagram.
+
+Similarly, the following sequence diagram shows how the
+`AnswerTracker` stores all the user answer inputs:  
+![AnswerTracker sequence diagram](https://via.placeholder.com/100.png?text=Photo)
+
+Step 2. The user may repeat Step 1 with other question sets.  
+
+The sequence diagram for this step involving `ResultsList` 
+and `AnswerTracker` are the same as the ones shown in Step 1.
+
+Step 3. The user now wants to view their results by executing
+the `results` command.  
+![Results command sequence diagram](https://via.placeholder.com/100.png?text=Photo)
+
+> **Note:** If the user uses the results feature before
+> attempting any questions, the application will instead 
+> return an error to the user indicating that there are no 
+> results.
+
+
+### Topics Feature
+
+The topics feature comprises `TopicList` and `QuestionListByTopic`. 
+`TopicList` is the list of topics for the users to attempt.
+`QuestionListByTopic` stores the respective question set for each topic in an ArrayList. 
+
+Given below is an example usage scenario and how the results
+mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time,
+and proceeds to start a game with their chosen topic.
+
+The following shows the class diagram for `topicList`:
```

Comment:
Perhaps add a line separator here to make the class diagram occupy a dedicated space would allow the format to be more readable.